src/Meshes/mesh.vertexData.ts

```diff
@@ -464,103 +464,121 @@ export class VertexData {
-        this.uvs4 = this._mergeElement(this.uvs4, other.uvs4);
-        this.uvs5 = this._mergeElement(this.uvs5, other.uvs5);
-        this.uvs6 = this._mergeElement(this.uvs6, other.uvs6);
-        this.colors = this._mergeElement(this.colors, other.colors);
-        this.matricesIndices = this._mergeElement(this.matricesIndices, other.matricesIndices);
-        this.matricesWeights = this._mergeElement(this.matricesWeights, other.matricesWeights);
-        this.matricesIndicesExtra = this._mergeElement(this.matricesIndicesExtra, other.matricesIndicesExtra);
-        this.matricesWeightsExtra = this._mergeElement(this.matricesWeightsExtra, other.matricesWeightsExtra);
+        this.positions = VertexData._mergeElement(this.positions, others.map(other => other.positions));
+        this.normals = VertexData._mergeElement(this.normals, others.map(other => other.normals));
+        this.tangents = VertexData._mergeElement(this.tangents, others.map(other => other.tangents));
+        this.uvs = VertexData._mergeElement(this.uvs, others.map(other => other.uvs));
+        this.uvs2 = VertexData._mergeElement(this.uvs2, others.map(other => other.uvs2));
+        this.uvs3 = VertexData._mergeElement(this.uvs3, others.map(other => other.uvs3));
+        this.uvs4 = VertexData._mergeElement(this.uvs4, others.map(other => other.uvs4));
+        this.uvs5 = VertexData._mergeElement(this.uvs5, others.map(other => other.uvs5));
+        this.uvs6 = VertexData._mergeElement(this.uvs6, others.map(other => other.uvs6));
+        this.colors = VertexData._mergeElement(this.colors, others.map(other => other.colors));
+        this.matricesIndices = VertexData._mergeElement(this.matricesIndices, others.map(other => other.matricesIndices));
+        this.matricesWeights = VertexData._mergeElement(this.matricesWeights, others.map(other => other.matricesWeights));
+        this.matricesIndicesExtra = VertexData._mergeElement(this.matricesIndicesExtra, others.map(other => other.matricesIndicesExtra));
+        this.matricesWeightsExtra = VertexData._mergeElement(this.matricesWeightsExtra, others.map(other => other.matricesWeightsExtra));
+
         return this;
     }
 
-    private _mergeElement(source: Nullable<FloatArray>, other: Nullable<FloatArray>): Nullable<FloatArray> {
-        if (!source) {
-            return other;
-        }
+    private static _mergeElement(source: Nullable<FloatArray>, others: readonly Nullable<FloatArray>[]): Nullable<FloatArray> {
+        const nonNullOthers = others.filter((other): other is FloatArray => other !== null && other !== undefined);
 
-        if (!other) {
+        if (nonNullOthers.length === 0) {
             return source;
         }
 
-        var len = other.length + source.length;
-        var isSrcTypedArray = source instanceof Float32Array;
-        var isOthTypedArray = other instanceof Float32Array;
+        if (!source) {
+            return this._mergeElement(nonNullOthers[1], nonNullOthers.slice(1));

```

Comment:
Shouldn't it be:
```typescript
            return this._mergeElement(nonNullOthers[0], nonNullOthers.slice(1));
```
?

Reply:
Wow how did I do that!? Glad you caught this one!

Reply:
@Popov72 King of reviews :-)